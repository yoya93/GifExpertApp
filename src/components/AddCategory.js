import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [imputValue, setImputValue] = useState("");

  const handleImputChange = (e) => {
    setImputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();

    if (imputValue.trim().length > 2) {
      setCategories((cate) => {
        setCategories([imputValue, ...cate]);
        setImputValue("");
      });
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <p>{imputValue}</p>
      <input type="text" value={imputValue} onChange={handleImputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
