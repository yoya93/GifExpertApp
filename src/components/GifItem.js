import React from "react";
import PropTypes from "prop-types";

export const GifItem = ({ title, url }) => {
  return (
    <div className="card  animate__heartBeat">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

// PrimeraApp.defaultProps = {
//   title: "titulo",
//   url: "https://youtube.com",
// };
