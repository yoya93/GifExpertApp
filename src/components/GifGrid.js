import React from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "./hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h2 className="animate__wobble">{category}</h2>

      {loading && <p> Loading</p>}

      <div className="card-grid">
        {images.map((img) => {
          return <GifItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
