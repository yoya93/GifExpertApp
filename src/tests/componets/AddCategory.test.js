import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import { AddCategory } from "../../components/AddCategory";

describe("pruebas en <AddCategory />", () => {
  const setCategories = jest.fn(); // tracking de la function
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    // reseteamos el componete despues de cada test
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe cambiar el input", () => {
    const input = wrapper.find("input");
    const value = "Hola Mundo";

    input.simulate("change", {
      target: {
        value: value,
      },
    });

    expect(wrapper.find("p").text().trim()).toBe(value);
  });

  test("debe disparar funcion setCategories", () => {
    const form = wrapper.find("form");

    const preventDefault = () => {};

    form.simulate("submit", { preventDefault });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("debe de llamar setCategories y limpiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Neymar Jr";

    input.simulate("change", {
      target: {
        value: value,
      },
    });

    const form = wrapper.find("form");
    const preventDefault = () => {};

    form.simulate("submit", { preventDefault });

    expect(setCategories).toHaveBeenCalledWith(expect.any(Function)); // asegurarme que se lla,a una funcion
    expect(setCategories).toHaveBeenCalled();
    expect(input.props().value).toBe("");
  });
});
