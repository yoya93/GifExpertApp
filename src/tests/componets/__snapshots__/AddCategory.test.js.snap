// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <AddCategory /> debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
